edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/DockerStats.cs

```diff
@@ -4,98 +4,170 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Docker.Models
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
     using Newtonsoft.Json;
 
     public class DockerStats
     {
+        [JsonConstructor]
+        public DockerStats(string name, Dictionary<string, DiskIO[]> block_io_stats, DockerCpuStats cpu_stats, MemoryStats memory_stats, Dictionary<string, NetworkInfo> networks, int? num_processes, PidsStats pids_stats, DateTime? read)
+        {
+            this.Name = Option.Maybe(name);
+            this.BlockIoStats = Option.Maybe(block_io_stats);
+            this.CpuStats = Option.Maybe(cpu_stats);
+            this.MemoryStats = Option.Maybe(memory_stats);
+            this.Networks = Option.Maybe(networks);
+            this.NumProcesses = Option.Maybe(num_processes);
+            this.PidsStats = Option.Maybe(pids_stats);
+            this.Read = Option.Maybe(read);
+
+            // Check for null entries in dictionarys
+            if (this.BlockIoStats.Exists(n => n.Any(o => o.Key == null || o.Value == null)))
+            {
+                this.BlockIoStats = Option.None<Dictionary<string, DiskIO[]>>();
```

Comment:
why not just filter out any value is null, but instead set to Option.None?

Reply:
switched to filter out instead